src/server/src/main/java/io/cassandrareaper/jmx/JmxProxyImpl.java

```diff
@@ -107,14 +108,14 @@
   private final EndpointSnitchInfoMBean endpointSnitchMbean;
   private final Object ssProxy;
   private final Object fdProxy;
-  private final Optional<RepairStatusHandler> repairStatusHandler;
   private final String host;
   private final String hostBeforeTranslation;
   private final JMXServiceURL jmxUrl;
   private final String clusterName;
+  private final ConcurrentMap<Integer, RepairStatusHandler> repairStatusHandlers =
+      Maps.newConcurrentMap();
```

Comment:
it would be good with a metrics gauge on this.

especially as i can see leaks happening: the call to `removeRepairStatusHandler(..)` is happening in a number of places outside of finally blocks.